### PR TITLE
Implement vector rotation

### PR DESCRIPTION
**Pull request by jeremystucki**

### CODE REVIEW

File: src/vector.rs [hunk 1]
```diff
@@ -571,4 +585,33 @@ mod tests {
...
+    #[test]
+    fn vector_rotated_by_half_pi_is_correct() {
+        let vector = Vector { x: 5.0, y: 10.0 };
+        let rotated_vector = vector.rotate(Radians::try_new(FRAC_PI_2).unwrap());
+
+        let expected_vector = Vector { x: -10.0, y: 5.0 };
+        assert_nearly_eq!(expected_vector.x, rotated_vector.x, 0.00001);
+        assert_nearly_eq!(expected_vector.y, rotated_vector.y, 0.00001);
+    }
 }
```
Comment (janhohenheim) [inline src/vector.rs:642]: Missing test cases: 
- Rotating by 2π and -2π
- Rotating twice by π
- Rotation in clockwise direction

File: src/vector.rs [hunk 2]
```diff
@@ -571,4 +585,33 @@ mod tests {
...
+    }
+
+    #[test]
+    fn vector_rotated_by_half_pi_is_correct() {
+        let vector = Vector { x: 5.0, y: 10.0 };
+        let rotated_vector = vector.rotate(Radians::try_new(FRAC_PI_2).unwrap());
+
+        let expected_vector = Vector { x: -10.0, y: 5.0 };
+        assert_nearly_eq!(expected_vector.x, rotated_vector.x, 0.00001);
+        assert_nearly_eq!(expected_vector.y, rotated_vector.y, 0.00001);
```
Comment (janhohenheim) [inline src/vector.rs]: Remove explicit precision

File: src/vector.rs [hunk 3]
```diff
@@ -126,12 +127,25 @@ impl Vector {
             other.unit() * self.dot_product(other) / other.magnitude()
         }
     }
+
+    /// Rotate a vector by the given amount (counterclockwise)
+    pub fn rotate(self, rotation: Radians) -> Vector {
+        let (rotation_sin, rotation_cos) = (-rotation.value()).sin_cos();
```
Comment (janhohenheim) [inline src/vector.rs]: Why is the negative of `rotation.value()` taken here? Leave a comment.

**Review by janhohenheim — CHANGES_REQUESTED** [review]

File: src/vector.rs (hunk 2 above)
Comment (jeremystucki) [reply src/vector.rs]: 4caf01f

File: src/vector.rs (hunk 1 above)
Comment (jeremystucki) [reply src/vector.rs:642]: 230a03c

Comment (jeremystucki) [reply src/vector.rs:642]: Note: Radians are only in the range 0 <= x < 2π

File: src/vector.rs (hunk 3 above)
Comment (jeremystucki) [reply src/vector.rs]: 1a63aa6

File: src/vector.rs (hunk 1 above)
Comment (janhohenheim) [reply src/vector.rs:642]: Remove code duplication and improve precision

**Review by janhohenheim — APPROVED** [review]

**Pull request merged**